### PR TITLE
Processing data in models

### PR DESCRIPTION
**Pull request by eroiyo**
Hello Microverse community, this is the milestone number 4, the next thing was implemented

- [ ] Models for Comments

- [ ] Models for Likes

- [ ] Models for Users

- [ ] Models for Posts

- [ ] Current_User function for the application controller implemented

### CODE REVIEW

**Review by ElijahTobs — CHANGES_REQUESTED** [review]
# 🚨 Invalid Code Review Request 🚨 
## Hi @eroiyo  👋 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

- According to your project specification, you're required to follow the *Gitflow* process but you didn't

_Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏_

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

**Comment by eroiyo** [comment]
Thanks, i i will fix the flow

File: app/models/comment.rb [hunk 1]
```diff
@@ -1,4 +1,7 @@
 # frozen_string_literal: true
 
 class Comment < ApplicationRecord
+  def update_counter(value)
+    post.update(comments_counter: value)
+  end
```
Comment (ElijahTobs) [inline app/models/comment.rb:6]: - [ ] Kindly set up associations between the four models in each of the __four files__ (`comment.rb, like.rb, user.rb, post.rb`)

File: app/models/post.rb [hunk 2]
```diff
@@ -1,4 +1,15 @@
...
 
 class Post < ApplicationRecord
+  def recent_comments(limit = 5)
+    comments.last limit
+  end
+
+  def update_counter(value)
+    user.update posts_counter: value
+  end
+
```
Comment (ElijahTobs) [inline app/models/post.rb:11]: - [ ] [__OPTIONAL__] You can as well use one of the [Rails Active Callbacks](https://guides.rubyonrails.org/active_record_callbacks.html) - `after_save` to call the `#update_post_counter` method every time a new post has been saved to database, then make the `update_post_counter` method private

**Review by ElijahTobs — COMMENTED** [review]
# Status: Changes Required ⚠️ 🚦
## Hi @eroiyo  👋

You've done really well so far! 👍 

## Major Highlights 💯 

You've followed the Gitflow rules. ✔️ 
You have good commit messages ✔️ 
Good PR title and summary ✔️ 

_However, there are some changes that you are required to make before this project can be approved._

## Required Changes ♻️


_See the inline comments below_

## Optional Changes 👌
_Every comment with the OPTIONAL prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend that you take them into account as they can make your code better._

- [ ] _Henceforth, kindly make sure you capitalize your commit messages_

_If anything doesn't seem clear to you, feel free to leave your question or comment in the PR thread below_

#### "Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise."

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
## Happy Coding :computer: 

_Reviewed by_  **Elijah O. Ayandokun**

**Comment by eroiyo** [comment]
Thanks i will do them

File: app/controllers/application_controller.rb [hunk 3]
```diff
@@ -1,4 +1,7 @@
 # frozen_string_literal: true
 
 class ApplicationController < ActionController::Base
+  def current_user
+    users.find(0)
+  end
```
Comment (DeliceLydia) [inline app/controllers/application_controller.rb:6]: ## Optional
- [ ] Here you can simply find the current_user by just doing `User.first` simple as that

**Review by DeliceLydia — APPROVED** [review]
# Congratulations🎉🎉

## Hi Team, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



### Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard).. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/controllers/application_controller.rb (hunk 3 above)
Comment (eroiyo) [reply app/controllers/application_controller.rb:6]: ooh, nice suggestion, thanks

**Pull request merged**